doc/how-to-source-packages.rst

```diff
@@ -0,0 +1,309 @@
+Hackage, on revisions not yet published or on forks.  This is the easiest way to
+work with a fork, much easier than using a Git submodule.  A dependency taken
+this way, effectively adds a local package to the project much like listing the
+package's source directory in the packages field, except that the source code is
+downloaded by Cabal using the version control system.
+
+For example, with a project that depends on the ``vector`` package from source
+we can see that the source code repository has been cloned by ``cabal build``:
+
+.. code-block:: text
+
+   $ cat cabal.project
+   ...
+   source-repository-package
+      type:     git
+      location: https://github.com/haskell/vector.git
+      subdir:   vector
+
+   $ cabal build all --dry-run
+   Cloning into '/.../dist-newstyle/src/vector-51ba9353b7a850a'...
+   remote: Enumerating objects: 9542, done.
+   remote: Counting objects: 100% (1800/1800), done.
+   remote: Compressing objects: 100% (652/652), done.
+   remote: Total 9542 (delta 1010), reused 1599 (delta 917), pack-reused 7742
+   Receiving objects: 100% (9542/9542), 2.52 MiB | 16.56 MiB/s, done.
+   Resolving deltas: 100% (5273/5273), done.
+   HEAD is now at 6b8bbc3 ...
+   Resolving dependencies...
+
+Cabal makes an archive from the cloned source code repository and uses this as
+the source for the dependency.
+
+.. code-block:: text
+
+      $ find -name '*.tar.gz'
+      ...
+      ./dist-newstyle/src/vector-51ba9353b7a850a-vector-0.13.1.0.tar.gz
+
+
+.. Warning::
+
+   The hash in the directory name of the clone is not the commit hash of the
+   cloned repository.
+
+   .. code-block:: text
+
+         $ cd dist-newstyle/src/vector-51ba9353b7a850a
+
+         $ git rev-parse HEAD
+         6b8bbc3a75b40451d8d225e30c576dfe89121c49
+
+*Source code* when dependency vendoring
+---------------------------------------
+
+If you want to include the source code of a package in your project, you can use
+the archive or the unpacked source code. This is called *vendoring* where you
+add the source code of an external package to your project.
+
+Vendoring a *source code* archive
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+After manually downloading a source code archive from Hackage (or elsewhere),
+you can add it to your project as a local package.
+
+   .. code-block:: shell
+
+      $ VER=0.13.1.0 \
+         curl -sSL https://hackage.haskell.org/package/vector-{$VER}/vector-{$VER}.tar.gz \
+         --output vector-{$VER}.tar.gz \
+      $ VER=0.13.1.0 echo "packages: vector-$VER.tar.gz" >> cabal.project
+      $ cabal build vector --dry-run
+      ...
+      Resolving dependencies...
+      Build profile: -w ghc-9.8.2 -O1
+      In order, the following would be built (use -v for more details):
+      - vector-0.13.1.0 (lib) (requires build)
+
+Vendoring unpacked *source code*
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You can use ``cabal get`` and then add the unpack directory to the ``packages``
```

Comment:
A ref to cabal get section would be good

Reply:
:+1: